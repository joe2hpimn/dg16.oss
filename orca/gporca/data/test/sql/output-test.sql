-- 
-- Table: r
-- 

drop table r;
create table r(a int, b int);

insert into r values (1 , 1);
insert into r values (2 , 1);
insert into r values (3 , 1);
insert into r values (4 , 1);
insert into r values (5 , 1);
insert into r values (5 , 1);
insert into r values (6 , 1);
insert into r values (6 , 3);
insert into r values (6 , 2);
insert into r values (7 , 1);
insert into r values (8 , 1);
insert into r values (9 , 2);
insert into r values (9 , 1);

-- 
-- Table: s
-- 

drop table s;
create table s(c int, d int, e int);

insert into s values (1 , 1 , 1);  
insert into s values (1 , 3 , 1);  
insert into s values (2 , 2 , 1);  
insert into s values (3 , 3 , 1);  
insert into s values (3 , 2 , 2);  
insert into s values (3 , 1 , 3);  
insert into s values (5 , 2 , 1);  
insert into s values (5 , 1 , 2);  
insert into s values (6 , 2 , 3);  
insert into s values (7 , 1 , 1);  
insert into s values (7 , 3 , 2);  
insert into s values (9 , 1 , 3);  
insert into s values (10 , 3 , 1);  
insert into s values (11 , 3 , 2);

analyze r;
analyze s;

--
-- Enable the optimizer
--

set gp_optimizer=on;

-- 
-- Display Table Contents
-- 

select * from r;
select * from s;

-- 
-- Query 1
-- 

select count(*) from r;

-- 
-- Query 2
-- 

select b, count(*), min(a), max(a), sum(a), min(c), max(c), sum(c) from r, s where r.b = s.d group by b order by b;

-- 
-- Query 3
-- 

select a, count(*), min(b), max(b), sum(b), min(d), max(d), sum(d) from r, s where r.a = s.c group by a order by a;

-- 
-- Query 4
-- 

select count(distinct a), sum(distinct a) from r;

-- 
-- Query 5
-- 

select count(distinct b), sum(distinct b) from r;

-- 
-- Query 6
-- 

select count(distinct b), sum(distinct b) from r group by a;

-- 
-- Query 7
-- 

select count(distinct a), sum(distinct a) from r group by b;

-- 
-- Query 8
-- 

select count(distinct c), sum(distinct c) from s group by d,e;

-- 
-- Query 9
-- 

select count(distinct a), sum(distinct a) as t from r order by t;

-- 
-- Query 10
-- 

select count(distinct b), sum(distinct b) as t from r group by a order by a,t ;

-- 
-- Query 11
-- 

select count(distinct a), sum(distinct a) as t from r group by b order by b, t;

-- 
-- Query 12
-- 

select count(distinct c), sum(distinct c) from s group by d,e order by e;

-- 
-- Query 13
-- 

select count(distinct c+d) from s;

-- 
-- Query 14
-- 

select count(distinct c+d) from s group by e;

-- 
-- Query 15
-- 

select count(distinct a), sum(distinct b) from r;

-- 
-- Query 16
-- 

select count(distinct c), sum(distinct d) from s;

-- 
-- Query 17
-- 

select count(distinct c), sum(distinct d) from s group by e;
