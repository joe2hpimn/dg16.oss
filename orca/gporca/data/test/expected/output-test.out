select pg_backend_pid();
 pg_backend_pid 
----------------
          37729
(1 row)

-- 
-- Table: r
-- 
drop table r;
DROP TABLE
create table r(a int, b int);
CREATE TABLE
insert into r values (1 , 1);
INSERT 0 1
insert into r values (2 , 1);
INSERT 0 1
insert into r values (3 , 1);
INSERT 0 1
insert into r values (4 , 1);
INSERT 0 1
insert into r values (5 , 1);
INSERT 0 1
insert into r values (5 , 1);
INSERT 0 1
insert into r values (6 , 1);
INSERT 0 1
insert into r values (6 , 3);
INSERT 0 1
insert into r values (6 , 2);
INSERT 0 1
insert into r values (7 , 1);
INSERT 0 1
insert into r values (8 , 1);
INSERT 0 1
insert into r values (9 , 2);
INSERT 0 1
insert into r values (9 , 1);
INSERT 0 1
-- 
-- Table: s
-- 
drop table s;
DROP TABLE
create table s(c int, d int, e int);
CREATE TABLE
insert into s values (1 , 1 , 1);  
INSERT 0 1
insert into s values (1 , 3 , 1);  
INSERT 0 1
insert into s values (2 , 2 , 1);  
INSERT 0 1
insert into s values (3 , 3 , 1);  
INSERT 0 1
insert into s values (3 , 2 , 2);  
INSERT 0 1
insert into s values (3 , 1 , 3);  
INSERT 0 1
insert into s values (5 , 2 , 1);  
INSERT 0 1
insert into s values (5 , 1 , 2);  
INSERT 0 1
insert into s values (6 , 2 , 3);  
INSERT 0 1
insert into s values (7 , 1 , 1);  
INSERT 0 1
insert into s values (7 , 3 , 2);  
INSERT 0 1
insert into s values (9 , 1 , 3);  
INSERT 0 1
insert into s values (10 , 3 , 1);  
INSERT 0 1
insert into s values (11 , 3 , 2);
INSERT 0 1
--
-- Enable the optimizer
--
set gp_optimizer=on;
SET
-- 
-- Display Table Contents
-- 
select * from r;
 a | b 
---+---
 1 | 1
 3 | 1
 5 | 1
 5 | 1
 7 | 1
 9 | 2
 9 | 1
 2 | 1
 4 | 1
 6 | 1
 6 | 3
 6 | 2
 8 | 1
(13 rows)

select * from s;
 c  | d | e 
----+---+---
  2 | 2 | 1
  6 | 2 | 3
 10 | 3 | 1
  1 | 1 | 1
  1 | 3 | 1
  3 | 3 | 1
  3 | 2 | 2
  3 | 1 | 3
  5 | 2 | 1
  5 | 1 | 2
  7 | 1 | 1
  7 | 3 | 2
  9 | 1 | 3
 11 | 3 | 2
(14 rows)

-- 
-- Query 1
-- 
select count(*) from r;
 count 
-------
    13
(1 row)

-- 
-- Query 2
-- 
select b, count(*), min(a), max(a), avg(a), sum(a), min(c), max(c), sum(c), avg(c) from r, s where r.b = s.d group by b order by b;
 b | count | min | max | avg | sum | min | max | sum | avg 
---+-------+-----+-----+-----+-----+-----+-----+-----+-----
 1 |    50 |   1 |   9 |   5 | 250 |   1 |   9 | 250 |   5
 2 |     8 |   6 |   9 | 7.5 |  60 |   2 |   6 |  32 |   4
 3 |     5 |   6 |   6 |   6 |  30 |   1 |  11 |  32 | 6.4
(3 rows)

-- 
-- Query 3
-- 
select a, count(*), min(b), max(b), avg(b), sum(b), min(d), max(d), sum(d), avg(d) from r, s where r.a = s.c group by a order by a;
 a | count | min | max | avg | sum | min | max | sum | avg 
---+-------+-----+-----+-----+-----+-----+-----+-----+-----
 1 |     2 |   1 |   1 |   1 |   2 |   1 |   3 |   4 |   2
 2 |     1 |   1 |   1 |   1 |   1 |   2 |   2 |   2 |   2
 3 |     3 |   1 |   1 |   1 |   3 |   1 |   3 |   6 |   2
 5 |     4 |   1 |   1 |   1 |   4 |   1 |   2 |   6 | 1.5
 6 |     3 |   1 |   3 |   2 |   6 |   2 |   2 |   6 |   2
 7 |     2 |   1 |   1 |   1 |   2 |   1 |   3 |   4 |   2
 9 |     2 |   1 |   2 | 1.5 |   3 |   1 |   1 |   2 |   1
(7 rows)

-- 
-- Query 4
-- 
select count(distinct a), sum(distinct a) from r;
 count | sum 
-------+-----
     9 |  45
(1 row)

-- 
-- Query 5
-- 
select count(distinct b), sum(distinct b) from r;
 count | sum 
-------+-----
     3 |   6
(1 row)

-- 
-- Query 6
-- 
select count(distinct b), sum(distinct b) from r group by a;
 count | sum 
-------+-----
     1 |   1
     1 |   1
     3 |   6
     1 |   1
     1 |   1
     1 |   1
     1 |   1
     1 |   1
     2 |   3
(9 rows)

-- 
-- Query 7
-- 
select count(distinct a), sum(distinct a) from r group by b;
 count | sum 
-------+-----
     2 |  15
     9 |  45
     1 |   6
(3 rows)

-- 
-- Query 8
-- 
select count(distinct c), sum(distinct c) from s group by d,e;
 count | sum 
-------+-----
     1 |   5
     2 |   7
     1 |   6
     2 |  18
     2 |   8
     2 |  12
     1 |   3
     3 |  14
(8 rows)

-- 
-- Query 9
-- 
select count(distinct a), sum(distinct a) as t from r order by t;
 count | t  
-------+----
     9 | 45
(1 row)

-- 
-- Query 10
-- 
select count(distinct b), sum(distinct b) as t from r group by a order by a,t ;
 count | t 
-------+---
     1 | 1
     1 | 1
     1 | 1
     1 | 1
     1 | 1
     3 | 6
     1 | 1
     1 | 1
     2 | 3
(9 rows)

-- 
-- Query 11
-- 
select count(distinct a), sum(distinct a) as t from r group by b order by b, t;
 count | t  
-------+----
     9 | 45
     2 | 15
     1 |  6
(3 rows)

-- 
-- Query 12
-- 
select count(distinct c), sum(distinct c) from s group by d,e order by e;
 count | sum 
-------+-----
     2 |   7
     3 |  14
     2 |   8
     1 |   3
     2 |  18
     1 |   5
     1 |   6
     2 |  12
(8 rows)

-- 
-- Query 13
-- 
select count(distinct c+d) from s;
 count 
-------
     9
(1 row)

-- 
-- Query 14
-- 
select count(distinct c+d) from s group by e;
 count 
-------
     4
     6
     3
(3 rows)

-- 
-- Query 15
-- 
select count(distinct a), sum(distinct b) from r;
 count | sum 
-------+-----
     9 |   6
(1 row)

-- 
-- Query 16
-- 
select count(distinct c), sum(distinct d) from s;
 count | sum 
-------+-----
     9 |   6
(1 row)

-- 
-- Query 17
-- 
select count(distinct c), sum(distinct d) from s group by e;
 count | sum 
-------+-----
     4 |   6
     3 |   3
     6 |   6
(3 rows)

